flume-hdfs-sink/src/main/java/com/wipro/ats/bdre/flume/sink/hdfs/BDREHDFSEventSink.java

```diff
@@ -445,21 +434,35 @@ public void run(String bucketPath) {
       }
     } catch (IOException eIO) {
       transaction.rollback();
-      LOG.warn("HDFS IO error", eIO);
+      LOGGER.warn("HDFS IO error", eIO);
       return Status.BACKOFF;
-    } catch (Throwable th) {
+    } catch (Exception ex) {
       transaction.rollback();
-      LOG.error("process failed", th);
-      if (th instanceof Error) {
-        throw (Error) th;
-      } else {
-        throw new EventDeliveryException(th);
-      }
+      LOGGER.error("process failed", ex);
+      throw new EventDeliveryException(ex);
     } finally {
       transaction.close();
     }
   }
 
+  @SuppressWarnings("squid:S1166")
+  private void writeTheDataToHDFS(Event event, String realPath, String realName, String lookupPath, BucketWriter bucketWriter, WriterCallback closeCallback) throws IOException, InterruptedException {
+    HDFSWriter hdfsWriter;
+    try {
+      bucketWriter.append(event);
+    } catch (BucketClosedException ex) {
+      LOGGER.info("Bucket was closed while trying to append, " +
+        "reinitializing bucket and writing event.");
+      hdfsWriter = writerFactory.getWriter(fileType);
+      bucketWriter = initializeBucketWriter(realPath, realName,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "bucketWriter". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1226)